extension/src/ALObject/ALObject2.ts

```diff
@@ -0,0 +1,283 @@
+                if (currObject === null) {
+                    throw new Error(`File '${objectFileName}' does not have valid object names. Maybe it got double quotes (") in the object name?`);
+                }
+                objectId = ALObject2.GetObjectId(currObject[2]);
+                objectName = currObject[3];
+                extendedObjectId = ALObject2.GetObjectId(currObject[6] ? currObject[6] : '');
+                extendedObjectName = ALObject2.TrimAndRemoveQuotes(currObject[4]);
+                extendedTableId = ALObject2.GetObjectId(currObject[8] ? currObject[8] : '');
+
+                break;
+            }
+
+            case ALObjectType.Profile:
+            case ALObjectType.Interface:
+                {
+
+                    const objectDescriptorPattern = new RegExp('(\\w+)( +"?[ a-zA-Z0-9._/&-]+"?)');
+                    let currObject = objectDescriptorCode.match(objectDescriptorPattern);
+                    if (currObject === null) {
+                        throw new Error(`File '${objectFileName}' does not have valid object names. Maybe it got double quotes (") in the object name?`);
+                    }
+
+                    objectId = 0;
+                    objectName = currObject[2];
+
+                    break;
+                }
+            case ALObjectType.PageCustomization: {
+
+                const objectDescriptorPattern = new RegExp('(\\w+)( +"?[ a-zA-Z0-9._/&-]+"?) +customizes( +"?[ a-zA-Z0-9._&-]+\\/?[ a-zA-Z0-9._&-]+"?) (\\/\\/+ *)?([0-9]+)?');
+                let currObject = objectDescriptorCode.match(objectDescriptorPattern);
+                if (currObject === null) {
+                    throw new Error(`File '${objectFileName}' does not have valid object names. Maybe it got double quotes (") in the object name?`);
+                }
+
+                objectId = 0;
+                objectName = currObject[2];
+
+                break;
+            }
+            default: {
+                Error(`Unhandled object type '${objectType}'`);
+            }
+        }
+
+
+
+        objectName = ALObject2.TrimAndRemoveQuotes(objectName);
+        return {
+            objectType: objectType,
+            objectId: objectId,
+            objectName: objectName,
+            extendedObjectId: extendedObjectId,
+            extendedObjectName: extendedObjectName,
+            extendedTableId: extendedTableId,
+            objectDescriptorLineNo: objectDescriptorLineNo
+
+        };
+    }
+
+
+    private static getObjectTypeArr(objectText: string) {
+        const objectTypePattern = new RegExp('(codeunit |page |pagecustomization |pageextension |profile |query |report |requestpage |table |tableextension |xmlport |enum |enumextension |interface )', "i");
+
+        return objectText.match(objectTypePattern);
+    }
+
+    private static getObjectType(objectTypeText: string, fileName?: string): ALObjectType {
+        switch (objectTypeText.trim().toLowerCase()) {
```

Comment:
Couldn't this switch be an preloaded object or map? It would make for some prettier code. 
```TypeScript
const typeText = objectTypeText.trim().toLowerCase();
if (ALObjects.hasOwnProperty(typeText) {
   return AlObjects[typeText];
} else if (filename) {
    Error(`Unknown object type ${typeText} in file ${fileName}`);
} else {
   Error(`Unknown object type ${typeText}`);
}
```

Reply:
since the objectTypeText could be wrongly cased I don't know if that would work... 

Wouldn't `AlObjectType['pAge']` fail?

Since TypeScript is case sensitive, I assumed that it would fail, but I haven't investigated it....

Reply:
one solution would be to let all enum values being UPPERCASE. Then we could transform the parameter to upper and it would work...

Reply:
Yes, `AlObjectType['pAge']` do fail.

Seems to be a solution to it, see https://stackoverflow.com/a/64932909.
But I don't know if it's worth it...?

Reply:
Either I'm missing something or we have a misunderstanding. Yes casing would brake the key, but you're already handling that, my suggestion wouldn't handle that much different? :) 

```TypeScript
// Map or object declared elsewere, outside of function
const ALObjectTypeMap = new Map<string, ALObjectType>();
ALObjectTypeMap.set('page', ALObjectType.Page);
ALObjectTypeMap.set('codeunit', ALObjectType.Codeunit);
ALObjectTypeMap.set('query', ALObjectType.Query);
ALObjectTypeMap.set('report', ALObjectType.Report);
ALObjectTypeMap.set('requestpage', ALObjectType.RequestPage);
ALObjectTypeMap.set('table', ALObjectType.Table);
ALObjectTypeMap.set('xmlport', ALObjectType.XmlPort);
ALObjectTypeMap.set('enum', ALObjectType.Enum);
ALObjectTypeMap.set('pageextension', ALObjectType.PageExtension);
ALObjectTypeMap.set('tableextension', ALObjectType.TableExtension);
ALObjectTypeMap.set('enumextension', ALObjectType.EnumExtension);
ALObjectTypeMap.set('profile', ALObjectType.Profile);
ALObjectTypeMap.set('interface', ALObjectType.Interface);
ALObjectTypeMap.set('pagecustomization', ALObjectType.PageCustomization);

// Same logic but less verbose
private static getObjectType(objectTypeText: string, fileName?: string): ALObjectType {
    const typeText = objectTypeText.trim().toLowerCase();
    if (ALObjectTypeMap.has(typeText) {
       return ALObjectTypes[typeText];
    } else if (filename) {
        Error(`Unknown object type ${typeText} in file ${fileName}`);
    } else {
       Error(`Unknown object type ${typeText}`);
   }
}

```

Reply:
Now I understand your point! 😁